sparse_strips/vello_common/src/encode.rs

```diff
@@ -375,15 +403,15 @@ fn encode_stops(stops: &[ColorStop], start: f32, end: f32, pad: bool) -> Vec<Gra
             factors[i] = c1_minus_c0 / x1_minus_x0;
         }
 
-        GradientRange {
+        vec![GradientRange {
             x0,
             x1,
             c0,
             factors,
-        }
+        }]
     };
 
-    let stop_ranges = stops.windows(2).map(|s| {
+    let stop_ranges = interpolated_stops.windows(2).map(|s| {
```

Comment:
```suggestion
    // Note: this could use `Iterator::map_windows` once stabilized, meaning `interpolated_stops`
    // no longer needs to be collected.
    let stop_ranges = interpolated_stops.windows(2).map(|s| {
```

Once [`Iterator::map_windows`](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.map_windows) is stabilized, this would no longer need to have an allocated `interpolated_stops`, as it no longer needs `slice::windows`.

On the other hand, as a `Vec` is being built by this function (and not an iterator), the function could also be rewritten to just be a loop over `stops`, rather than building an explicit iterator that is collected at the end.

Reply:
(This might be a nice candidate for [generators](https://rust-lang.github.io/rfcs/3513-gen-blocks.html).)